src/import/prepare.rs

```diff
@@ -25,13 +24,41 @@ use crate::util::rocksdb_util::properties::SizeProperties;
 use super::client::*;
 use super::common::*;
 use super::engine::*;
+use super::metrics::*;
 use super::{Config, Error, Result};
 
 const MAX_RETRY_TIMES: u64 = 3;
 const RETRY_INTERVAL_SECS: u64 = 1;
 
-const SCATTER_WAIT_MAX_RETRY_TIMES: u64 = 125;
-const SCATTER_WAIT_INTERVAL_MILLIS: u64 = 8;
+const SPLIT_WAIT_MAX_RETRY_TIMES: u64 = 64;
+const SPLIT_WAIT_INTERVAL_MILLIS: u64 = 8;
+const SPLIT_MAX_WAIT_INTERVAL_MILLIS: u64 = 1000;
+const SCATTER_WAIT_MAX_RETRY_TIMES: u64 = 128;
+const SCATTER_WAIT_INTERVAL_MILLIS: u64 = 50;
+const SCATTER_MAX_WAIT_INTERVAL_MILLIS: u64 = 5000;
+
+macro_rules! exec_with_retry {
+    ($tag:expr, $func:expr, $times:expr, $interval:expr, $max_duration:expr) => {
+        let start = Instant::now();
+        for i in 0..$times {
+            if $func {
+                if i > 0 {
+                    debug!(concat!("waited between ", $tag); "retry times" => %i, "takes" => ?start.elapsed());
+                }
+                break;
+            } else if i == $times - 1 {
+                warn!(concat!($tag, " still failed after exhausting all retries"));
+            } else {
+                // Exponential back-off with max wait duration
+                let mut interval = $interval * (1 << (i as u64));
```

Comment:
It is **invalid** to write `1 << i` when `i > 64`. Perhaps just keep `interval` as a variable outside the for loop and repeatedly perform `interval *= 2`.